### PR TITLE
Scraper modulation

### PR DESCRIPTION
**Pull request by nicolassaw**
Changes made:
 - made the scraper more modular by breaking functions into smaller methods
 - made the methods pure functions to not require dependencies from the class
 - removed the 'test' parameter and instead passed the necessary dependencies from the unit test so there was no test logic in the prod code

### CODE REVIEW

File: src/parser/__init__.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                    # Handle the case where parser_instance or parser_function is None
+                    print("Error: Could not obtain parser instance or function.")
+
+                # Adds county field to data
+                case_data['county'] = county
+
+                # Adds a hash to the JSON file of the underlying HTML
+                body = case_soup.find("body")
+                balance_table = body.find_all("table")[-1]
+                if "Balance Due" in balance_table.text:
```
Comment (normaljosh) [inline src/parser/__init__.py:168]: Let's add a comment in the code explaining why we do this 

(The "balance due" part of the html changes every day you view the case file, even if nothing else about the case changes. So we don't want that to be part of the unique text that we hash to check if anything has been changed.)

**Review by normaljosh — APPROVED** [review]
Good work, I think seperating all this out into classes will make for a more scaleable codebase! 

Most of my comments are somewhat nitpicky, from what I can tell this will still run which is the most important part!

File: src/parser/__init__.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+import sys
+import importlib
+from bs4 import BeautifulSoup
+
+class Parser:
+
+    def __init__(self):
+        pass
+
+    def get_class_and_method(self, county):
```
[the remark this answers is not in the document]
Comment (nicolassaw) [reply src/parser/__init__.py:16]: Interesting! I'll turn my head around this a bit more and think about how I might implement it. Want to make sure I understand this better. Thanks so much!

File: src/parser/__init__.py (hunk 1 above)
Comment (nicolassaw) [reply src/parser/__init__.py:168]: Great point! I'll put this in the next update.

**Pull request merged**